code/game/machinery/overwatch.dm

```diff
@@ -820,110 +699,392 @@
+	if(current_squad.primary_objective)
+		dat += "[current_squad.primary_objective]<br>"
+	else
+		dat += "<b><font color=red>NONE!</font></b><br>"
+	dat += "<b>Secondary Objective:</b> "
+	if(current_squad.secondary_objective)
+		dat += "[current_squad.secondary_objective]<br>"
+	else
+		dat += "<b><font color=red>NONE!</font></b><br>"
+	dat += get_squad_info_ending()
+	return dat
+
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/get_squad_info_ending()
+	var/dat = ""
+	dat += "----------------------<br>"
+	dat += "<A href='?src=[REF(src)];operation=refresh'>{Refresh}</a><br>"
+	dat += "<A href='?src=[REF(src)];operation=change_sort'>{Change Sorting Method}</a><br>"
+	dat += "<A href='?src=[REF(src)];operation=hide_dead'>{[dead_hidden ? "Show Dead Marines" : "Hide Dead Marines" ]}</a><br>"
+	dat += "<A href='?src=[REF(src)];operation=choose_z'>{Change Locations Ignored}</a><br>"
+	dat += "<br><A href='?src=[REF(src)];operation=back'>{Back}</a>"
+	return dat
+
+
+/obj/machinery/computer/camera_advanced/overwatch/main
+	icon_state = "overwatch_main"
+	name = "Main Overwatch Console"
+	desc = "State of the art machinery for general overwatch purposes."
+
+
+
+/obj/machinery/computer/camera_advanced/overwatch/main/attack_hand(mob/user)
+	. = ..()
+	if(.)  //Checks for power outages
+		return
+	if(!allowed(user))
+		to_chat(user, "<span class='warning'>You don't have access.</span>")
+		return
+	if(!length(squads))
+		for(var/i in SSjob.squads)
+			var/datum/squad/S = SSjob.squads[i]
+			squads += S
+	user.set_interaction(src)
+	var/dat
+	if(!operator)
+		dat += "<B>Main Operator:</b> <A href='?src=[REF(src)];operation=change_main_operator'>----------</A><BR>"
+	else
+		dat += "<B>Main Operator:</b> <A href='?src=[REF(src)];operation=change_main_operator'>[operator.name]</A><BR>"
+		dat += "   <A href='?src=[REF(src)];operation=logout_main'>{Stop Overwatch}</A><BR>"
+		dat += "----------------------<br>"
+		switch(state)
+			if(OW_MAIN)
+				for(var/datum/squad/S in squads)
+					dat += "<b>[S.name] Squad</b> <a href='?src=[REF(src)];operation=message;current_squad=[REF(S)]'>\[Message Squad\]</a><br>"
+					if(S.squad_leader)
+						dat += "<b>Leader:</b> <a href='?src=[REF(src)];operation=use_cam;cam_target=[REF(S.squad_leader)]'>[S.squad_leader.name]</a> "
+						dat += "<a href='?src=[REF(src)];operation=sl_message;current_squad=[REF(S)]'>\[MSG\]</a><br>"
+					else
+						dat += "<b>Leader:</b> <font color=red>NONE</font><br>"
+					if(S.overwatch_officer)
+						dat += "<b>Squad Overwatch:</b> [S.overwatch_officer.name]<br>"
+					else
+						dat += "<b>Squad Overwatch:</b> <font color=red>NONE</font><br>"
+					dat += "<A href='?src=[REF(src)];operation=monitor[S.id]'>[S.name] Squad Monitor</a><br>"
+				dat += "----------------------<br>"
+				dat += "<b>Orbital Bombardment Control</b><br>"
+				dat += "<b>Current Cannon Status:</b> "
+				var/cooldown_left = (almayer_orbital_cannon.last_orbital_firing + 5000) - world.time
```

Comment:
```suggestion
				var/cooldown_left = (almayer_orbital_cannon.last_orbital_firing + 8 MINUTES) - world.time
```

Reply:
This one would require the other `5000` to be changed as well. Technically it would be `((8 MINUTES) + (20 SECONDS))`, but don't think it's really important to keep the seconds.